cmake/modules/shields.cmake

```
@@ -79,41 +79,65 @@ foreach(root ${BOARD_ROOT})
 
       list(REMOVE_ITEM SHIELD-NOTFOUND ${s})
 
-      # Add <shield>.overlay to the shield_dts_files output variable.
+      # Add <shield>.overlay to a temporary list
       list(APPEND
-        shield_dts_files
+        shield_${s}_dts_files
         ${SHIELD_DIR_${s}}/${s}.overlay
         )
 
-      # Add the shield's directory to the SHIELD_DIRS output variable.
+      # Add the shield's directory to a temporary list
       list(APPEND
-        SHIELD_DIRS
+        shield_${s}_dir
         ${SHIELD_DIR_${s}}
         )
```

similar, there can only be a single dir for a given shield.
```suggestion
      set(shield_${s}_dir ${SHIELD_DIR_${s}})
```

---

Fixed